### PR TITLE
single server procfile: postgres_exporter respects env vars

### PR DESCRIPTION
**Pull request by uwedeportivo**
Fixes https://github.com/sourcegraph/sourcegraph/issues/6735

### CODE REVIEW

File: cmd/server/shared/shared.go [hunk 1]
```diff
@@ -126,6 +126,11 @@ func Main() {
 		log.Fatal("Failed to setup nginx:", err)
 	}
 
+	postgresEporter, err := postgresExporterProcFile()
```
Comment (slimsag) [inline cmd/server/shared/shared.go]: typo? `Eporter` vs `Exporter`

Comment (uwedeportivo) [reply cmd/server/shared/shared.go]: done

**Comment by keegancsmith** [comment]
Sorry I missed this! We provide a function which does exactly this, and covers all the corner cases you might have missed. https://sourcegraph.com/github.com/sourcegraph/sourcegraph@v3.10.0/-/blob/cmd/frontend/internal/cli/config.go#L261

**Comment by uwedeportivo** [comment]
perfect, sounds good. i'll send a follow-up PR that uses it

**Comment by uwedeportivo** [comment]
https://github.com/sourcegraph/sourcegraph/pull/6838

**Pull request merged**